src/Umbraco.Core/Deploy/ArtifactBase.cs

```diff
@@ -1,61 +1,50 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Runtime.Serialization;
 
 namespace Umbraco.Cms.Core.Deploy
 {
     /// <summary>
     /// Provides a base class to all artifacts.
     /// </summary>
-    [DataContract]
     public abstract class ArtifactBase<TUdi> : IArtifact
         where TUdi : Udi
     {
         protected ArtifactBase(TUdi udi, IEnumerable<ArtifactDependency> dependencies = null)
         {
-            if (udi == null)
-                throw new ArgumentNullException("udi");
-            Udi = udi;
+            Udi = udi ?? throw new ArgumentNullException("udi");
             Name = Udi.ToString();
 
             Dependencies = dependencies ?? Enumerable.Empty<ArtifactDependency>();
             _checksum = new Lazy<string>(GetChecksum);
         }
 
         private readonly Lazy<string> _checksum;
+
         private IEnumerable<ArtifactDependency> _dependencies;
 
         protected abstract string GetChecksum();
 
         #region Abstract implementation of IArtifactSignature
 
-        Udi IArtifactSignature.Udi
-        {
-            get { return Udi; }
-        }
+        Udi IArtifactSignature.Udi => Udi;
 
-        [DataMember]
         public TUdi Udi { get; set; }
 
-        [IgnoreDataMember]
-        public string Checksum
-        {
-            get { return _checksum.Value; }
-        }
+        public string Checksum => _checksum.Value;
+
+        public bool ShouldSerializeChecksum() => false;
```

Comment:
We need a comment on this one pointing to some kind of docs, like https://www.newtonsoft.com/json/help/html/ConditionalProperties.htm

I think this is the only place in the entire codebase we use this functionality

Reply:
I've added a comment referencing this.